src/query/api/v1/handler/prometheus/native/read_test.go

```diff
@@ -116,13 +119,51 @@ func TestPromReadHandlerWithTimeout(t *testing.T) {
 		Params:    r,
 	}
 
-	_, err := read(ctx, parsed, promRead.opts)
+	_, err := execRead(ctx, parsed, promRead.opts)
 	require.Error(t, err)
 	require.Equal(t,
 		"context deadline exceeded",
 		err.Error())
 }
 
+func BenchmarkReadHandlerAllocs(b *testing.B) {
```

Comment:
Adding this benchmark to the bottom of the test which is what I've seen in other places in the repo. For posterity, allocs are roughly the same in old vs new (93 vs 97):

```
# pre-change
goos: darwin
goarch: amd64
pkg: github.com/m3db/m3/src/query/api/v1/handler/prometheus/native
BenchmarkReadHandlerAllocs
BenchmarkReadHandlerAllocs-12    	   14320	     81337 ns/op	   14535 B/op	      93 allocs/op
PASS

# post-change
goos: darwin
goarch: amd64
pkg: github.com/m3db/m3/src/query/api/v1/handler/prometheus/native
BenchmarkReadHandlerAllocs
BenchmarkReadHandlerAllocs-12    	   11882	     97325 ns/op	   15352 B/op	      97 allocs/op
PASS
```